pipeline/scripts/tda/CE_percentile_rank_analysis.py

```diff
@@ -1,41 +1,79 @@
+    rank_data = data.rank(axis=1, method="min", na_option="bottom")
 
-# Compute the 90th percentile for each column in the DataFrame
-percentile = rank_data.quantile(q=0.9)
+    # Divide by the number of non-NA values for each row to normalize
+    rank_data = rank_data.div(data.notna().sum(axis=1), axis=0)
 
-# Calculate the density function of the 90th percentile values
-dens = pd.Series(percentile).plot.density(bw_method=0.3).get_lines()[0].get_xydata()
-dens_df = pd.DataFrame(dens, columns=["x", "y"])
+    # Compute the 90th percentile for each column in the DataFrame
+    percentile = rank_data.quantile(q=0.9)
 
-# Filter out the values that are not between 0.1 and 0.9
-df_range = dens_df[(dens_df["x"] > 0.1) & (dens_df["x"] < 0.9)]
+    # Calculate the density function of the 90th percentile values
+    dens = pd.Series(percentile).plot.density(bw_method=0.3).get_lines()[0].get_xydata()
+    dens_df = pd.DataFrame(dens, columns=["x", "y"])
 
-# Find the x value that corresponds to the minimum y value in the filtered DataFrame
-threshold = df_range.loc[df_range["y"].idxmin(), "x"]
-ce_list = pd.DataFrame(
-    {"Row.name": percentile.index, "CE_percentile": percentile.values}
-)
+    # Filter out the values that are not between 0.1 and 0.9
+    df_range = dens_df[(dens_df["x"] > 0.1) & (dens_df["x"] < 0.9)]
 
-# Add a boolean column to the DataFrame that indicates whether the CE_percentile value is <= to the threshold
-ce_list["Common_Essential"] = ce_list["CE_percentile"] <= threshold
+    # Find the x value that corresponds to the minimum y value in the filtered DataFrame
+    threshold = df_range.loc[df_range["y"].idxmin(), "x"]
+    ce_df = pd.DataFrame(
+        {"Row.name": percentile.index, "CE_percentile": percentile.values}
+    )
 
-ce_list.to_csv(out_file, index=False)
+    # Add a boolean column to the DataFrame that indicates whether the CE_percentile value is <= to the threshold
+    ce_df["Common_Essential"] = ce_df["CE_percentile"] <= threshold
+
+    ce_df.to_csv(out_file, index=False)
+
+    return ce_df
+
+
+def get_common_essential_genes_for_chronos_combined(
+    crispr_inferred_ce_file: str, out_file: str
+) -> pd.DataFrame:
+    """
+    Get the common essential genes for Chronos_Combined and format it so that 
+    it can be merged with the other data
+    """
+    crispr_inferred_ce_df = pd.read_csv(crispr_inferred_ce_file)
+    crispr_inferred_ce_df.rename(columns={"Essentials": "Row.name"}, inplace=True)
+    crispr_inferred_ce_df["Common_Essential"] = True
+
+    crispr_inferred_ce_df.to_csv(out_file, index=False)
+
+    return crispr_inferred_ce_df
+
+
+if __name__ == "__main__":
+    import sys
+
+    data_label = sys.argv[1]
+    crispr_inferred_common_essentials_file = sys.argv[2]
+    non_crispr_input_data_file = sys.argv[3]
+    out_file = sys.argv[4]
+
+    # Crispr is retreived from taiga
+    # Non-crispr is calculated from the data file
+    if data_label == "Chronos_Combined":
```

Comment:
I personally would have gone the route of having two different conseq rules: one for RNAi (which would be the original rule, running the original script) and one for Chronos_Combined which reformats the file. 

It feels clearer to me. For example, if I understand how this will work, when run on RNAi, the rule is going to have a dependency on crispr_inferred_common_essentials_file even when that file is ignored.

Anyway, I don't think it's worth changing anything because I think this is a change that is largely isolated from anything that'll we'll be touching anytime soon.

Just sharing a different approach.